test/integration-no-mock/config-management/env-config-push.js

```diff
@@ -0,0 +1,172 @@
+      android: {
+        senderId: 'id123',
+        apiKey: 'key123'
+      }
+    };
+
+    const groups = {
+      oneGroup: {
+        description: 'One group',
+        groups: ['otherGroup']
+      },
+      otherGroup: {}
+    };
+
+    const endpoints = {
+      myEndpoint: {
+        type: 'internal',
+        code: 'function onRequest(request, response, modules) {\nconsole.log("On request...");\n  response.continue();\n}'
+      }
+    };
+
+    const initialEnv = {
+      schemaVersion: '1.0.0',
+      configType: 'environment',
+      collections: ConfigManagementHelper.common.buildConfigEntityFromList(collList),
+      roles: ConfigManagementHelper.common.buildConfigEntityFromList(rolesList),
+      push: pushSettings,
+      customEndpoints: endpoints,
+      groups
+    };
+
+    // modified env
+    const roleName = rolesList[0].name;
+    const collWithComplexPermissions = collList[0];
+    collWithComplexPermissions.permissions = {
+      'all-users': {
+        create: 'always',
+        read: 'grant',
+        update: 'always',
+        delete: 'grant'
+      },
+      [roleName]: {
+        create: 'grant',
+        update: 'entity'
+      }
+    };
+
+    const collHooks = {
+      [collList[0].name]: {
+        onPreSave: {
+          type: 'internal'
+        }
+      },
+      [collList[1].name]: {
+        onPostFetch: {
+          type: 'internal',
+          code: 'function onPostFetch(request, response, modules) {\nconsole.log("Here");\n  response.continue();\n}'
+        }
+      }
+    };
+
+    const modifiedEnv = cloneDeep(initialEnv);
+    modifiedEnv.collections = ConfigManagementHelper.common.buildConfigEntityFromList(collList);
+    modifiedEnv.collectionHooks = collHooks;
+    modifiedEnv.push.android.apiKey = '456';
+    modifiedEnv.groups.otherGroup = { name: 'Other group', description: 'Description for other group' };
+    modifiedEnv.customEndpoints.myEndpoint.schedule = {
+      start: moment().add(1, 'month').toISOString()
+    };
+
+    const modifiedConfig = cloneDeep(modifiedEnv);
+    delete modifiedConfig.roles;
+
+    const envName = randomStrings.envName();
+    const rolesNameIdPairs = {};
+    let envId;
+
+    async.series([
+      (next) => {
+        ConfigManagementHelper.env.createFromConfig(envName, initialEnv, appName, (err, id) => {
+          if (err) {
+            return next(err);
+          }
+
+          envId = id;
+          next();
+        });
+      },
+      (next) => {
+        ConfigManagementHelper.env.modifyFromConfig(modifiedConfig, envName, appName, next);
+      },
+      (next) => {
+        ConfigManagementHelper.env.assertEnvOnly(modifiedEnv, envId, envName, next);
+      },
+      (next) => {
+        ApiService.roles.get(envId, null, (err, actualRoles) => {
+          if (err) {
+            return next(err);
+          }
+
+          const expectedRolesCount = rolesList.length;
+          expect(expectedRolesCount).to.equal(actualRoles.length);
```

Comment:
Expected and Actual should be exchanged.

Reply:
I caught myself several times while writing the tests thinking which goes first and which second :grin: Will fix.